src/go/k8s/pkg/resources/statefulset_scale.go

```diff
@@ -0,0 +1,392 @@
+	r.logger.Info("Handling cluster in recommissioning phase")
+
+	// First we ensure we've enough replicas to let the recommissioning node run
+	targetReplicas := *r.pandaCluster.Status.DecommissioningNode + 1
+	err := setCurrentReplicas(ctx, r, r.pandaCluster, targetReplicas, r.logger)
+	if err != nil {
+		return err
+	}
+
+	adminAPI, err := r.getAdminAPIClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	broker, err := getNodeInfoFromCluster(ctx, *r.pandaCluster.Status.DecommissioningNode, adminAPI)
+	if err != nil {
+		return err
+	}
+
+	if broker == nil || broker.MembershipStatus != admin.MembershipStatusActive {
+		err = adminAPI.RecommissionBroker(ctx, int(*r.pandaCluster.Status.DecommissioningNode))
+		if err != nil {
+			return fmt.Errorf("error while trying to recommission node %d in cluster %s: %w", *r.pandaCluster.Status.DecommissioningNode, r.pandaCluster.Name, err)
+		}
+		r.logger.Info("Node marked for being recommissioned in cluster", "node_id", *r.pandaCluster.Status.DecommissioningNode)
+
+		return &RequeueAfterError{
+			RequeueAfter: wait.Jitter(r.decommissionWaitInterval, decommissionWaitJitterFactor),
+			Msg:          fmt.Sprintf("Waiting for node %d to be recommissioned into cluster %s", *r.pandaCluster.Status.DecommissioningNode, r.pandaCluster.Name),
+		}
+	}
+
+	r.logger.Info("Recommissioning process successfully completed", "node_id", *r.pandaCluster.Status.DecommissioningNode)
+	r.pandaCluster.Status.DecommissioningNode = nil
+	return r.Status().Update(ctx, r.pandaCluster)
+}
+
+func (r *StatefulSetResource) getAdminAPIClient(
+	ctx context.Context, ordinals ...int32,
+) (adminutils.AdminAPIClient, error) {
+	return r.adminAPIClientFactory(ctx, r, r.pandaCluster, r.serviceFQDN, r.adminTLSConfigProvider, ordinals...)
+}
+
+func (r *StatefulSetResource) isClusterFormed(
+	ctx context.Context,
+) (bool, error) {
+	rootNodeAdminAPI, err := r.getAdminAPIClient(ctx, 0)
+	if err != nil {
+		return false, err
+	}
+	brokers, err := rootNodeAdminAPI.Brokers(ctx)
+	if err != nil {
+		// Eat the error and return that the cluster is not formed
+		return false, nil
+	}
+	return len(brokers) > 0, nil
+}
+
+// disableMaintenanceModeOnDecommissionedNodes can be used to put a cluster in a consistent state, disabling maintenance mode on
+// nodes that have been decommissioned.
+//
+// A decommissioned node may activate maintenance mode via shutdown hooks and the cluster may enter an inconsistent state,
+// preventing other pods clean shutdown.
+//
+// See: https://github.com/redpanda-data/redpanda/issues/4999
+func (r *StatefulSetResource) disableMaintenanceModeOnDecommissionedNodes(
+	ctx context.Context,
+) error {
+	if !featuregates.MaintenanceMode(r.pandaCluster.Status.Version) {
+		return nil
+	}
+
+	if r.pandaCluster.Status.DecommissioningNode == nil || r.pandaCluster.Status.CurrentReplicas > *r.pandaCluster.Status.DecommissioningNode {
+		// Only if actually in a decommissioning phase
```

Comment:
NIT: This comment miss-leads as it is the opposite.  When entered it means do nothing as decommissioning is not executed or it is scaling up. 

This comment would be more accurate 2 lines down.

Reply:
```suggestion
		// Only if not in decommissioning phase
```